### PR TITLE
[Term Entry] Swift Protocols Decodable #2958

### PR DESCRIPTION
**Pull request by sdwinston**
### Description

-Added Decodable Protocol term entry for Swift
-Closes Issue #2958

### Type of Change

- Adding a new entry

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [ ] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

File: content/swift/concepts/protocols/terms/decodable/decodable.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+Tags:
+  - 'Swift'
+  - 'iOS'
+  - 'JSON'
+CatalogContent:
+  - 'learn-swift'
+  - 'paths/build-ios-apps-with-swiftui'
+---
+
+`Decodable` is a type in Swift that can convert itself into a structure that conforms to the **`Decodable protocol`**. The `Decodable protocol` is used to easily decode JSON or similar data formats like XML. The required initializer is `init(from: Decoder)` which creates a new instance by decoding from the given decoder. The `Decodable protocol` is often used with the `JSONDecoder` class. `Decodable` should be used if you only need to read data.
```
Comment (SSwiniarski) [inline content/swift/concepts/protocols/terms/decodable/decodable.md:17]:
```suggestion
`Decodable` is a type in Swift that can convert itself into a structure that conforms to the **`Decodable`** protocol. The `Decodable` protocol is used to easily decode JSON or similar data formats like XML. The required initializer is `init(from: Decoder)` which creates a new instance by decoding from the given decoder. The `Decodable` protocol is often used with the `JSONDecoder` class. `Decodable` should be used when there's only a need to read data. 
```

File: content/swift/concepts/protocols/terms/decodable/decodable.md [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+  - 'JSON'
+CatalogContent:
+  - 'learn-swift'
+  - 'paths/build-ios-apps-with-swiftui'
+---
+
+`Decodable` is a type in Swift that can convert itself into a structure that conforms to the **`Decodable protocol`**. The `Decodable protocol` is used to easily decode JSON or similar data formats like XML. The required initializer is `init(from: Decoder)` which creates a new instance by decoding from the given decoder. The `Decodable protocol` is often used with the `JSONDecoder` class. `Decodable` should be used if you only need to read data. 
+
+## Syntax
+Create a structure to represent the data to decode, followed by `Decodable` using the standard syntax of a Swift protocol.
```
Comment (SSwiniarski) [inline content/swift/concepts/protocols/terms/decodable/decodable.md:20]:
```suggestion
## Syntax

Create a structure to represent the data to decode, followed by `Decodable` using the standard syntax of a Swift protocol.   
```

File: content/swift/concepts/protocols/terms/decodable/decodable.md [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+Create a structure to represent the data to decode, followed by `Decodable` using the standard syntax of a Swift protocol.   
+
+```pseudo
+struct DataName: Decodable {
+  // Properties that conform to Decodable
+}
+```
+
+## Example
+In the example below, a structure called User is created and the `Decodable` protocol is used with `JSONDecoder` to convert JSON stored as a data object.
```
Comment (SSwiniarski) [inline content/swift/concepts/protocols/terms/decodable/decodable.md:29]:
```suggestion

In the example below, a structure called User is created and the `Decodable` protocol is used with `JSONDecoder` to convert stored JSON into a data object.
```

File: content/swift/concepts/protocols/terms/decodable/decodable.md [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+""".data(using: .utf8)!
+
+let decoder = JSONDecoder()
+let user = try decoder.decode(User.self, from: json)
+print(user.name) // Prints "Charlotte Lucas"
+
+```
+
+
+
```
Comment (SSwiniarski) [inline content/swift/concepts/protocols/terms/decodable/decodable.md:52]:
````suggestion
```
````
Remove extra blank lines.

**Review by SSwiniarski — CHANGES_REQUESTED** [review]
@sdwinston I did a first review and left a few comments. Also note you still need to sign the [CLA](https://cla-assistant.io/Codecademy/docs?pullRequest=3198).

**Review by SSwiniarski — APPROVED** [review]
@sdwinston This looks good for a second review.

File: content/swift/concepts/protocols/terms/decodable/decodable.md [hunk 5]
```diff
@@ -0,0 +1,50 @@
+---
+Title: 'Decodable'
+Description: 'A Swift protocol that enables the conversion of data.'
+Subjects: 
+  - 'iOS'
+  - 'Mobile Development'
+  - 'Computer Science'
```
Comment (caupolicandiaz) [inline content/swift/concepts/protocols/terms/decodable/decodable.md:7]:
```suggestion
```

Comment (caupolicandiaz) [reply content/swift/concepts/protocols/terms/decodable/decodable.md:7]: we limit subjects to one or two items

File: content/swift/concepts/protocols/terms/decodable/decodable.md [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+Tags:
+  - 'Swift'
+  - 'iOS'
+  - 'JSON'
+CatalogContent:
+  - 'learn-swift'
+  - 'paths/build-ios-apps-with-swiftui'
+---
+
+`Decodable` is a type in Swift that can convert itself into a structure that conforms to the **`Decodable`** protocol. The `Decodable` protocol is used to easily decode JSON or similar data formats like XML. The required initializer is `init(from: Decoder)` which creates a new instance by decoding from the given decoder. The `Decodable` protocol is often used with the `JSONDecoder` class. `Decodable` should be used when there's only a need to read data.
```
Comment (caupolicandiaz) [inline content/swift/concepts/protocols/terms/decodable/decodable.md:17]:
```suggestion
The **`Decodable`** protocol is used to easily decode JSON or similar data formats like XML. The required initializer is `init(from: Decoder)` which creates a new instance by decoding from the given decoder. The `Decodable` protocol is often used with the `JSONDecoder` class. `Decodable` should be used when there's only a need to read data, when reading and writing are required the `Codable` protocol should be implemented.
```

File: content/swift/concepts/protocols/terms/decodable/decodable.md [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+
+```pseudo
+struct DataName: Decodable {
+  // Properties that conform to Decodable
+}
+```
+
+## Example
+
+In the example below, a structure called User is created and the `Decodable` protocol is used with `JSONDecoder` to convert stored JSON into a data object.
```
Comment (caupolicandiaz) [inline content/swift/concepts/protocols/terms/decodable/decodable.md:31]:
```suggestion
In the example below, a `User` struct is defined and the `Decodable` protocol is used with `JSONDecoder` to convert stored JSON into a data object.
```

File: content/swift/concepts/protocols/terms/decodable/decodable.md [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+let json = """ {
+  "name": "Charlotte Lucas"
+  "age": 27
+  "location": "England"
+}
+""".data(using: .utf8)!
+
+let decoder = JSONDecoder()
+let user = try decoder.decode(User.self, from: json)
+print(user.name) // Prints "Charlotte Lucas"
```
Comment (caupolicandiaz) [inline content/swift/concepts/protocols/terms/decodable/decodable.md:49]:
```suggestion
import Foundation
struct User: Decodable {
   var name: String
   var age: Int
   var location: String
 }

 let userJson = """
 {
   "name": "Charlotte Lucas",
   "age": 27,
   "location": "England"
 }
 """
 
 let userData = Data(userJson.utf8)

 let decoder = JSONDecoder()
 let user = try decoder.decode(User.self, from: userData)
 print(user.name)
```

Comment (caupolicandiaz) [reply content/swift/concepts/protocols/terms/decodable/decodable.md:49]: we usually use a ```shell block for the output

**Review by caupolicandiaz — CHANGES_REQUESTED** [review]
@sdwinston I've added a few additional comments. Please update the entry, and tag me when you've finished making changes. Thanks

**Comment by sdwinston** [comment]
@caupolicandiaz I've updated the entry with the changes

**Review by caupolicandiaz — APPROVED** [review]
@sdwinston @SSwiniarski LGTM

**Pull request merged**